File: frontend/transport-management-system-web/src/app/types/location.ts
// location.ts
export interface ILocation {
  id: number;
  name: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
  address?: string;
  latitude?: number;
  longitude?: number;
  zoneName?: string;
  geographicalEntities?: ILocationGeographicalEntity[];
}

export interface ILocationGeographicalEntity {
  geographicalEntityId: number;
  name?: string;
}

export interface ICreateLocationDto {
  name: string;
  geographicalEntities?: ILocationGeographicalEntity[];
  isActive?: boolean;
}

export interface IUpdateLocationDto {
  name?: string;
  geographicalEntities?: ILocationGeographicalEntity[];
  isActive?: boolean;
}
export interface ApiResponse<T> {
  success: boolean;
  message: string;
  data: T;
}

File: backend/TransportManagementSystem/Entity/Location.cs
using System.ComponentModel.DataAnnotations;

namespace TransportManagementSystem.Entity;

public class Location
{
    public int Id { get; set; }

    [Required]
    public string Name { get; set; } = string.Empty;

    public string? AddressText { get; set; }
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }
    public bool IsValidated { get; set; } = false;

    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    [Required]
    public string Address { get; set; } = string.Empty;
}
